### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by dnikolic98**

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -4,7 +4,7 @@ PODS:
     - Kingfisher/Core (= 5.14.1)
   - Kingfisher/Core (5.14.1)
   - PureLayout (3.1.6)
-  - ReachabilitySwift (5.0.0)
+  - ReachabilitySwift (4.3.1)
```
Comment (labasbozidar) [inline Podfile.lock:7]: You can remove this because you are using RxReachability.

File: weatherApp/App/Data/WeatherRepository.swift [hunk 2]
```diff
@@ -136,4 +142,11 @@ class WeatherRepository {
         }
     }
     
+    private func bindReachability() {
+        Reachability.rx
```
Comment (labasbozidar) [inline weatherApp/App/Data/WeatherRepository.swift]: Expose this as property:
```
let isReachable: Bool {
   return Reachability.rx.isReachable
}
```

File: weatherApp/App/LocationService/LocationService.swift [hunk 3]
```diff
@@ -28,8 +29,31 @@ class LocationService: LocationServiceProtocol {
                 return Coordinates(latitude: coord.latitude, longitude: coord.longitude)
             }
         
+        isEnabled = Observable
+            .combineLatest(
+                locationManager.rx.isEnabled,
+                locationManager.rx.didChangeAuthorization)
+            .flatMap { enabled, _  -> Observable<Bool> in
```
Comment (labasbozidar) [inline weatherApp/App/LocationService/LocationService.swift:36]: .flatMap { $0 }

File: weatherApp/App/UI/Views/UserWarningView+Design.swift [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        addSubview(warningInfo)
+    }
+    
+    func styleViews() {
+        styleView()
+        styleWarningLabel()
+    }
+    
+    func defineLayoutForViews() {
+        warningInfo.autoPinEdge(.leading, to: .leading, of: self, withOffset: 15)
```
Comment (labasbozidar) [inline weatherApp/App/UI/Views/UserWarningView+Design.swift]: Use constant for this value.

File: weatherApp/Features/WeatherDetails/UI/DetailWeatherPresenter.swift [hunk 5]
```diff
@@ -43,6 +43,12 @@ class DetailWeatherPresenter {
         return fiveDaysList
     }
     
+    func isReachable() -> Observable<Bool> {
```
Comment (labasbozidar) [inline weatherApp/Features/WeatherDetails/UI/DetailWeatherPresenter.swift]: You can use property for this.

File: weatherApp/Features/WeatherDetails/UI/DetailWeatherViewController.swift [hunk 6]
```diff
@@ -22,11 +22,14 @@ class DetailWeatherViewController: UIViewController {
     private var dataDisposeBag: DisposeBag = DisposeBag()
     private var timerDisposeBag: DisposeBag = DisposeBag()
     private var refreshDisposeBag: DisposeBag = DisposeBag()
+    private var reachableDisposeBag: DisposeBag = DisposeBag()
```
Comment (labasbozidar) [inline weatherApp/Features/WeatherDetails/UI/DetailWeatherViewController.swift]: Please check if you need all of these dispose bags.

File: weatherApp/Features/WeatherDetails/UI/DetailWeatherViewController.swift [hunk 7]
```diff
@@ -137,6 +141,34 @@ class DetailWeatherViewController: UIViewController {
...
+                guard reachable else {
+                    self.showInternetWarning()
+                    return
+                }
+                self.hideInternetWarning()
+            })
+            .disposed(by: reachableDisposeBag)
+    }
+    
+    private func showInternetWarning() {
```
Comment (labasbozidar) [inline weatherApp/Features/WeatherDetails/UI/DetailWeatherViewController.swift]:
```
private func showInternetWarning(_ showWarning: Bool) { ... }
```

File: weatherApp/Features/WeatherDetails/UI/DetailWeatherViewController.swift [hunk 8]
```diff
@@ -137,6 +141,34 @@ class DetailWeatherViewController: UIViewController {
...
+        DispatchQueue.main.async {
```
Comment (labasbozidar) [inline weatherApp/Features/WeatherDetails/UI/DetailWeatherViewController.swift]: Replace this with .observeOn(Main)

File: weatherApp/Features/WeatherList/UI/CurrentWeatherListPresenter.swift [hunk 9]
```diff
@@ -73,6 +73,21 @@ class CurrentWeatherListPresenter {
             })
     }
     
+    func isReachable() -> Observable<Bool> {
```
Comment (labasbozidar) [inline weatherApp/Features/WeatherList/UI/CurrentWeatherListPresenter.swift]: These three functions can be properties.

File: weatherApp/Features/WeatherList/UI/CurrentWeatherListPresenter.swift [hunk 10]
```diff
@@ -73,6 +73,21 @@ class CurrentWeatherListPresenter {
...
+        weatherRepository
+            .isReachable
+            .skip(1)
+            .asObservable()
+    }
+    
+    func checkLocationsAllowed() -> Bool {
```
Comment (labasbozidar) [inline weatherApp/Features/WeatherList/UI/CurrentWeatherListPresenter.swift]: Rename to areLocationsAllowed.

**Review by labasbozidar — CHANGES_REQUESTED** [review]

File: Podfile.lock (hunk 1 above)
Comment (dnikolic98) [reply Podfile.lock:7]: Happens it gets installed with RxReachability

File: weatherApp/App/LocationService/LocationService.swift (hunk 3 above)
Comment (dnikolic98) [reply weatherApp/App/LocationService/LocationService.swift:36]: Xcode doesn't like it that way
```
Type '(Bool, ControlEvent<CLAuthorizationEvent>.Element)' (aka '(Bool, (manager: CLLocationManager, status: CLAuthorizationStatus))') cannot conform to 'ObservableConvertibleType'; only struct/enum/class types can conform to protocols
```

File: weatherApp/Features/WeatherList/UI/HomeViewController.swift [hunk 11]
```diff
@@ -102,6 +108,57 @@ class HomeViewController: UIViewController {
...
+                }
+                
+                let allowed = self.currentWeatherListPresenter.areLocationsAllowed
+                self.showLocationsWarning(!allowed, warning: LocalizedStrings.authLocationsWarning)
+            })
+            .disposed(by: viewControllerDisposeBag)
+    }
+    
+    private func showLocationsWarning(_ showWarning: Bool, warning: String) {
+        switch showWarning {
```
Comment (labasbozidar) [inline weatherApp/Features/WeatherList/UI/HomeViewController.swift]: Use if else for bool property.

File: weatherApp/Features/WeatherDetails/UI/DetailWeatherViewController.swift [hunk 12]
```diff
@@ -122,6 +125,28 @@ class DetailWeatherViewController: UIViewController {
...
+            .isReachable
+            .observeOn(MainScheduler.instance)
+            .subscribe(onNext: { [weak self] reachable in
+                self?.showInternetWarning(!reachable)
+            })
+            .disposed(by: viewControllerDisposeBag)
+    }
+    
+    private func showInternetWarning(_ showWarning: Bool) {
+        switch showWarning {
```
Comment (labasbozidar) [inline weatherApp/Features/WeatherDetails/UI/DetailWeatherViewController.swift]: You don't need to use switch for bool value. Please use if condition.

**Review by labasbozidar — CHANGES_REQUESTED** [review]

**Review by labasbozidar — APPROVED** [review]

**Pull request merged**